src/main/java/seedu/address/model/event/UniqueEventList.java

```diff
@@ -1,18 +1,84 @@
+
 import java.util.Iterator;
 
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
+import seedu.address.model.event.exceptions.DuplicateEventException;
+import seedu.address.model.event.exceptions.EventNotFoundException;
 
+/**
+ * A list of events that enforces uniqueness between its elements and does not allow nulls.
+ * An event is considered unique using {@code Event#isSameEvent}. As such, adding and updating of
+ * events uses #Event#isSameEvent(Event) for equality so as to ensure the event being added is
+ * unique in terms of identity in the UniqueEventList. However, the removal of an event uses Event#equals so
+ * as to ensure that the event with the exact same fields will be removed.
+ */
 public class UniqueEventList implements Iterable<Event> {
-    //TODO: [Event] add functions
 
     private final ObservableList<Event> internalList =
-            FXCollections.observableArrayList(new Event("Hall dinner"), new Event("Hall supper"));
+            FXCollections.observableArrayList();
     private final ObservableList<Event> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
 
+    /**
+     * Creates a dummy implementation for UniqueEventList.
+     */
+    public UniqueEventList() {
+        Event dummy1 = new Event(new EventName("Hall supper"), new Description("Have supper together"));
+        Event dummy2 = new Event(new EventName("Hall dinner"), new Description("Have dinner together"));
+
+        internalList.add(dummy1);
+        internalList.add(dummy2);
+    }
+
+    /**
+     * Returns true if the list contains an equivalent event in the given argument.
+     */
+    public boolean contains(Event toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameEvent);
+    }
+
+    /**
+     * Adds an event to the list.
+     * The event must not already exist in the list.
+     */
+    public void add(Event toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateEventException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent event from the list.
+     * The event must exist in the list.
+     */
+    public void remove(Event toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new EventNotFoundException();
+        }
+    }
+
+    /**
+     * Replaces the event {@code target} in the list with {@code editedEvent}.
+     * {@code target} must exist in the list.
+     * The event identity of {@code editedPerson} must not be the same as another existing event in the list.
+     */
+    public void setEvent(Event target, Event editedEvent) {
+        requireAllNonNull(target, editedEvent);
+
+        int index = internalList.indexOf(target);
+
+        internalList.set(index, editedEvent);
+    }
```

Comment:
Changes made here.

Reply:
Okay for this is fine for now, but maybe you can check if the `editedEvent` and `target` events are unique before setting them into the list? Else you might get duplicates within the `UniqueEventList`.

Reply:
So do an equality check? i.e `target.equals(editedEvent)` 
I was thinking of doing it in the `AssignCommand#execute()`

Reply:
Can you do it in both? So its similar to `UniquePersonList.java#setPerson`

Reply:
Ok will do